cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -525,3 +536,38 @@ func (c *CStorPoolController) getDeviceIDs(csp *apis.CStorPool) ([]string, error
 	// TODO: Add error handling
 	return pool.GetDeviceIDs(csp)
 }
+
+func (c *CStorPoolController) upgrade(csp *apis.CStorPool) (*apis.CStorPool, error) {
+	if csp.VersionDetails.Current != csp.VersionDetails.Desired &&
+		csp.VersionDetails.Desired != "" {
+		// As no other steps are required just change current version to
+		// desired version
+		csp.VersionDetails.Current = csp.VersionDetails.Desired
+		obj, err := c.clientset.OpenebsV1alpha1().CStorPools().Update(csp)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to update SPC with versionDetails")
+		}
+		return obj, nil
+	}
+	return csp, nil
+}
+
+// populateVersion assigns VersionDetails for old csp object
+func (c *CStorPoolController) populateVersion(csp *apis.CStorPool) (
+	*apis.CStorPool, error,
+) {
+	v := csp.Labels[string(apis.OpenEBSVersionKey)]
+	// 1.3.0 onwards new CSP will have the field populated during creation
+	if v < "1.3.0" && csp.VersionDetails.Current == "" {
```

Comment:
does '<' works with strings?

Reply:
yes it will start from left most character of both strings and take the ascii difference of the two characters.